gen/enum_test.go

```diff
@@ -399,3 +399,69 @@ func TestEnumAccessors(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumLabelLegit(t *testing.T) {
+	tests := []struct {
+		enumItem  te.EnumWithLabel
+		wireValue string
+	}{
+		{te.EnumWithLabelUsername, "\"surname\""},
+		{te.EnumWithLabelPassword, "\"hashed_password\""},
+		{te.EnumWithLabelSalt, "\"salt\""},
+		{te.EnumWithLabelSugar, "\"sugar\""},
+	}
+	for _, tt := range tests {
+		t.Run("compare label:"+tt.wireValue, func(t *testing.T) {
+			// marshal
+			label := te.EnumWithLabel(tt.enumItem)
+			b, err := json.Marshal(label)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.wireValue, string(b))
+
+			// unmarshal
+			var expectedLabel te.EnumWithLabel
+			err = json.Unmarshal(b, &expectedLabel)
+			assert.Equal(t, expectedLabel, label)
+
+			assertRoundTrip(
+				t, &tt.enumItem,
+				wire.NewValueI32(int32(tt.enumItem)),
+				"test roundtrip "+tt.wireValue,
+			)
+		})
+	}
+}
+
+func TestEnumLabelInvalid(t *testing.T) {
+
+	for _, tt := range []struct {
+		value te.EnumWithLabel
+		rep   string
+	}{
+		{-1, "-1"},
+		{1 << 10, "1024"},
+	} {
+		invalidEnumItem := te.EnumWithLabel(tt.value)
+		b, err := json.Marshal(invalidEnumItem)
+		assert.NoError(t, err)
```

Comment:
this is default marshal behavior: if not matching any key, just marshal as string representation of int value, without raising error